### PR TITLE
[circle-partitioner] Introduce HelperPath

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce HelperPath with path related helper functions.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #6230, from draft #6231

File: compiler/circle-partitioner/src/HelperPath.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  return base.substr(pos + 1);
+}
+
+std::string make_path(const std::string &base, const std::string &input, uint32_t idx,
+                      const std::string &backend)
+{
+  auto filename_ext = get_filename_ext(input);
+
+  auto pos = filename_ext.find_last_of(".");
+  assert(pos > 0);
```
Comment (mhs4670go) [inline compiler/circle-partitioner/src/HelperPath.cpp:50]: This function assumes length of pos is always positive?

Comment (seanshpark) [reply compiler/circle-partitioner/src/HelperPath.cpp:50]: This line seems strange. Thanks for point out. I think some revisition is required...

**Review by mhs4670go — APPROVED** [review]
LGTM

**Pull request merged**